test/e2e/nsx_inventory_test.go

```diff
@@ -0,0 +1,324 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestInventorySync(t *testing.T) {
```

Comment:
Do we need to test the ContainerCluster?

Reply:
added